Implement record diff snapshot request event

corovcam opened the pull request:
Related Linear Issue: https://linear.app/ducesnet/issue/FE-384/implement-record-diff-snapshot-request-event

Ducica commented:
![image](https://github.com/user-attachments/assets/49972192-fc2f-4079-bbe7-ae96e88da684)

Ducica commented:
I would consider making the entire event "foldable", meaning that you can fold the table, because it could get quite long and maybe it could become an obstacle if person is there for comments or other things - please double check  with @mirekys

mirekys commented:
<!-- linear:replyTo:2954877362 -->

Agreed, it should be wrapped under accordion component, also fileld label text shouldn be wrapped on spaces, not splitting words

Ducica commented:
Maybe to clear out space, and to have less problems with column sizes, it would be enough to color green or red the "Pole" column text, or potentially Stara hodnota/Nova hodnota and therefore eliminate "Operace" column.

corovcam commented:
> ![image](https://private-user-images.githubusercontent.com/34883098/452899091-49972192-fc2f-4079-bbe7-ae96e88da684.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDk0ODE5NjAsIm5iZiI6MTc0OTQ4MTY2MCwicGF0aCI6Ii8zNDg4MzA5OC80NTI4OTkwOTEtNDk5NzIxOTItZmMyZi00MDc5LWJiZTctYWU5NmU4OGRhNjg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjA5VDE1MDc0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwN2VhNWRhZDBmMDYwNThhMTA3YWYyNGZhZmUwMzJhMzBhMDQ4NjQ3MTNiNzEyYTViZmU2ZGRkZWU4NzhjNzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.BylSfG1MV4CvpORlMrKGVBbobbJrjcji9JsD7sbO8l4)
> 
> I would recommend to try to make the minimal record and then when Editing metadata, fill the entire record, it would probably be a good indication of where else there could be such issues ([object object])
> 
> Note: this screenshot is when you are viewing the request in the modal.

> Also when not in modal, there are some slight issues as you reduce the screen size: [Screencast from 2025-06-09 09-04-28.webm](https://github.com/user-attachments/assets/3e93cfdd-9fc1-4d59-8856-eb3a42464b53)

> Agreed, it should be wrapped under accordion component, also fileld label text should be wrapped on spaces, not splitting words

Thanks for feedback. I agree the CSS was problematic. Here is the current version:
![image](https://github.com/user-attachments/assets/575eeef4-8afa-4a58-8c7d-c908971c9aae)

> Yes the column seem to be redundant, it can be just removed, and field label color coded red/green/yellow

I don't think it would be easily distinguishable for different operation types what the color means. To save space, I rather added an Icon with help-text (title) as the first column.

![image](https://github.com/user-attachments/assets/44b8e938-fe09-496d-a92f-63182df4f1d5)

Ducica commented:
Thanks a lot for the fixes. From my side:

1. Please check the issue with styling, because now accordion appears in custom fields in Modal. Probably it can be solved by using some class name. If I remember correctly, I could not target that accordion better because it is inside of Invenio code
![image](https://github.com/user-attachments/assets/d0da0476-b09a-430a-86b9-62ab59504ed6)


2. I would still consider putting together that icon with field name, in the same column.

corovcam commented:
> Thanks a lot for the fixes. From my side:
> 
> 1. Please check the issue with styling, because now accordion appears in custom fields in Modal. Probably it can be solved by using some class name. If I remember correctly, I could not target that accordion better because it is inside of Invenio code
>    ![image](https://private-user-images.githubusercontent.com/34883098/453366519-d0da0476-b09a-430a-86b9-62ab59504ed6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDk4MTU2ODYsIm5iZiI6MTc0OTgxNTM4NiwicGF0aCI6Ii8zNDg4MzA5OC80NTMzNjY1MTktZDBkYTA0NzYtYjA5YS00MzBhLTg2YjktNjJhYjU5NTA0ZWQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjEzVDExNDk0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyZjdmYzJhYzA3ZWE1N2FjMmE0NmFlMjJjODRiY2JmMjNlMjAzZGRkZGZhMDI1MDlhZDFlMjdkNWVlOTQwYjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9PJ9IqkcpHS9NWxw7OY0gtlaEyWzOUfxrmwsrrk-yVM)
> 2. I would still consider putting together that icon with field name, in the same column.

1. Thanks. Fixed to not hide accordion title under .ui.feed only.

2. Ok, I just didn't want to break the normal form of table. to save one column in width, I changed it to this.
![image](https://github.com/user-attachments/assets/17685b90-1daa-426c-bdab-a6b9b3fcb056)

There is also a possibility of a more structured approach for the "Value" column splitting into "Old" and "New". I still find it necessary for these two to coexist cause it would be very unfriendly (and breaking relational normal forms) if the table combines "add", "delete" and "replace" operation into one "Value Diff " column (especially in the case of "replace" event).
![image](https://github.com/user-attachments/assets/af042b6f-075d-4696-a36d-a29e3ef2ad31)
![image](https://github.com/user-attachments/assets/cbd3caf9-1a2f-4802-a1bb-4608eb70302a)

corovcam commented:
> Thanks a lot for another iteration on this @corovcam. I seem to feel, based on this, that it makes sense to have 3 different "tables", withg actions grouped by operation type. The output would look like:
> 
> ## 🔻Show changes
> ### ➕ Added
> Field	Value
> `title -> cs`	cesky nazev
> `creatibutors -> 0 -> name`	Franta Novak
> ### ➖ Removed
> Field	Value
> `title -> cs`	smazany nazev
> `creatibutors -> 0 -> name`	Smazany creatibutor
> ### ✏️ Changed
> Field	Old value	New value
> `title -> cs`	Puvodni nazev	Novy nazev
> `creatibutors -> 0 -> name`	Puvodni creatibutor	Novy creatibutor
> ~ I won't go with nested table headers here -- takes much vertical space, and it might be less responsive ~
> 
> Each group/table should be rendered only if some action of that type exists... What do you guys think? :)

Okay thanks, this might save some space and looks similar to some websites I've seen, e.g. forums, github release changes, etc.

mirekys commented:
Pls also rebase (to fix conflicts) :)

corovcam commented:
Current implementation:

![image](https://github.com/user-attachments/assets/868beb63-edb6-44fc-86c4-ed6342dc339f)
By default it's hidden:
![image](https://github.com/user-attachments/assets/499b1c5c-be12-4944-b947-cc087a7ea828)

![image](https://github.com/user-attachments/assets/64f645c1-d533-4330-8fe3-fcb65f16c016)

mirekys commented on oarepo_requests/ui/theme/assets/semantic-ui/less/oarepo_requests_ui/themes/default/views/feed.overrides:
```diff
@@ -43,6 +43,99 @@
+              }
+              
+              &.replace {
+                border-left: 3px solid #f2711c !important;
```
move/merge to .replace rule block above

mirekys commented on oarepo_requests/ui/theme/assets/semantic-ui/js/oarepo_requests_ui/common/components/TimelineRecordDiffSnapshotEvent.jsx:
```diff
@@ -0,0 +1,222 @@
+        const isVisible = visibleAccordions[operationType] || false;
+
+        return (
+          <Accordion key={operationType} fluid className={`operation-accordion ${operationType}`}>
```
This calls for extraction into separate sub-component :) It would simplify TimelineRecordDiffSnapshotEvent, so that it doesn't have to keep track of what accordions are open
